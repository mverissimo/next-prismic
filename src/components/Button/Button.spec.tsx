import React from 'react';

import {
  create,
  render,
  fireEvent,
  renderToHtml,
  axe,
} from '@utils/test-utils';

import { Button } from '.';

describe('Button', () => {
  describe('styles', () => {
    it('should render with default style', () => {
      const actual = create(<Button>Button</Button>);

      expect(actual).toMatchSnapshot();
    });
  });

  describe('Behavior', () => {
    it('should accept a working ref for a Button', () => {
      const buttonRef = React.createRef<HTMLButtonElement>();
      const { container } = render(<Button ref={buttonRef}>Button</Button>);
      const button = container.querySelector('button');

      expect(buttonRef.current).toBe(button);
    });

    it('should call onClick handler when clicked', () => {
      const props = {
        onClick: jest.fn(),
        'data-testid': 'button',
      };
      const { getByTestId } = render(<Button {...props}>Button</Button>);

      fireEvent.click(getByTestId('button'));

      expect(props.onClick).toHaveBeenCalledTimes(1);
    });
  });

  describe('Acessibility', () => {
    it('should meet accessibility guidelines', async () => {
      const wrapper = renderToHtml(<Button>Button</Button>);
      const results = await axe(wrapper);

      expect(results).toHaveNoViolations();
    });
  });
});
