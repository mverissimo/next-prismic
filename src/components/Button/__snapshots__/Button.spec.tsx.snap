// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button styles should render with default style 1`] = `
.next-0 {
  color: #FFFFFF;
  border: 0;
  background: #000000;
}

.emotion-0 {
  color: #FFFFFF;
  border: 0;
  background: #000000;
}

<button
  class="emotion-0 emotion-1"
>
  Button
</button>
`;
